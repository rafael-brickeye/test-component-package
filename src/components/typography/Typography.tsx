import React from 'react';
import {Text, TextProps, TextStyle} from 'react-native';
import useStyles from './Typography.styles';

type TextDefaultProps = TextProps & TextStyle;

export type TypographyPropTypes = {
  truncateEnd?: boolean;
} & TextDefaultProps;

const Typography: React.FC<TypographyPropTypes> = ({children, truncateEnd = false, style: styleProps, ...props}) => {
  const style = useStyles({...props});

  return (
    <Text
      numberOfLines={truncateEnd ? 1 : 0}
      style={{...style.typographyContainer, ...(styleProps as TextStyle)}}
      {...props}>
      {children}
    </Text>
  );
};

export {Typography};
