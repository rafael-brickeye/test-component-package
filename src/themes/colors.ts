// All values used here come from Figma file design styling

const background = {
  white: '#ffffff',
};

const surface = {
  grey: '#C5CCD6',
};

const text = {
  black: '#000000DE',
  grey: '#71727A',
  white: '#ffffff',
};

const brand = {
  protect: {
    primary: {
      main: '#0048BB',
    },
  },
  lumicon: {
    primary: {
      light: '#FF9F34',
      main: '#FF8C4D',
      dark: '#FF5968',
    },
    secondary: {
      light: '#F46167',
      main: '#F05E40',
      dark: '#9E9E9E',
    },
    info: {
      light: '#64B6F7',
      main: '#2196F3',
      dark: '#0B79D0',
    },
    error: {
      light: '#F88078',
      main: '#F44336',
      dark: '#E31B0C',
    },
    warning: {
      light: '#FFB547',
      main: '#FF9800',
      dark: '#C77700',
    },
    success: {
      light: '#7BC67E',
      main: '#4CAF50',
      dark: '#3B873E',
    },
  },
};

const shadow = {
  default: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  wide: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4,
  },
};

export const color = {text, brand, surface, background, shadow};
