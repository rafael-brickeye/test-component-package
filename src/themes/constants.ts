import {TextStyle, ViewStyle} from 'react-native';

export const defaultViewStyles: Record<keyof Omit<ViewStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  height: 'height',
  justifyContent: 'justifyContent',
  left: 'left',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  zIndex: 'zIndex',
};

export const defaultTextStyles: Record<keyof Omit<TextStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  color: 'color',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  fontFamily: 'fontFamily',
  fontSize: 'fontSize',
  fontStyle: 'fontStyle',
  fontVariant: 'fontVariant',
  fontWeight: 'fontWeight',
  height: 'height',
  includeFontPadding: 'includeFontPadding',
  justifyContent: 'justifyContent',
  left: 'left',
  letterSpacing: 'letterSpacing',
  lineHeight: 'lineHeight',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  textAlign: 'textAlign',
  textAlignVertical: 'textAlignVertical',
  textDecorationColor: 'textDecorationColor',
  textDecorationLine: 'textDecorationLine',
  textDecorationStyle: 'textDecorationStyle',
  textShadowColor: 'textShadowColor',
  textShadowOffset: 'textShadowOffset',
  textShadowRadius: 'textShadowRadius',
  textTransform: 'textTransform',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  writingDirection: 'writingDirection',
  zIndex: 'zIndex',
};
