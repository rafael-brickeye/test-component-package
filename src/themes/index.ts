export * from './colors';
export * from './sizing';
export * from './spacing';
export * from './utils';
export * from './types';
export * from './constants';
