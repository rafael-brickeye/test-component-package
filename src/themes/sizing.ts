import {FontSizeTypes} from './types';
import {getFontSize} from './utils';

const font: Record<FontSizeTypes, number> = {
  xs: getFontSize(4),
  sm: getFontSize(5),
  md: getFontSize(6),
  lg: getFontSize(7),
  xl: getFontSize(8),
  '2xl': getFontSize(9),
  '3xl': getFontSize(10),
  '4xl': getFontSize(11),
  '5xl': getFontSize(12),
};

export const sizing = {font};
