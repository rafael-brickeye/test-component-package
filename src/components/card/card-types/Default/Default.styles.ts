import {StyleSheet} from 'react-native';
import {color, getDefaultStylesFromProps} from '../../../../themes';

const defaultStyles = {
  minHeight: 68,
  borderRadius: 5,
  backgroundColor: color.background.white,
  ...color.shadow.default,
};

const useStyles = ({...props}) => {
  const styles = getDefaultStylesFromProps(props);

  return StyleSheet.create({
    cardContainer: {
      ...defaultStyles,
      ...styles,
    },
  });
};

export default useStyles;
