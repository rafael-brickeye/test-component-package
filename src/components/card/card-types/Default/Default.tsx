import React from 'react';
import {TouchableHighlight, TouchableHighlightProps, ViewStyle} from 'react-native';
import useStyles from './Default.styles';
import {View} from '../../../view';
import {color} from '../../../../themes';

export type DefaultCardPropTypes = {
  onPress: () => void;
  isDisabled: boolean;
} & TouchableHighlightProps &
  ViewStyle;

const DefaultCard: React.FC<DefaultCardPropTypes> = ({isDisabled, onPress, children, style: styleProps, ...props}) => {
  const style = useStyles({isDisabled, ...props});

  return (
    <TouchableHighlight
      disabled={isDisabled}
      underlayColor={color.surface.grey}
      onPress={onPress}
      style={{...style.cardContainer, ...(styleProps as ViewStyle)}}
      {...props}>
      <View opacity={isDisabled ? 0.5 : 1}>{children}</View>
    </TouchableHighlight>
  );
};

export {DefaultCard};
