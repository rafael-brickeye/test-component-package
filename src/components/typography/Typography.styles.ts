import {StyleSheet} from 'react-native';
import {getDefaultStylesFromProps, sizing, color as themeColor} from '../../themes';

const useStyles = ({...props}) => {
  const styles = getDefaultStylesFromProps(props);

  return StyleSheet.create({
    typographyContainer: {
      fontSize: sizing.font.md,
      fontWeight: 'normal',
      color: themeColor.text.black,
      ...styles,
    },
  });
};

export default useStyles;
