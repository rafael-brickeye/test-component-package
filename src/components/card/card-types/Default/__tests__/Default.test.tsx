import React from 'react';
import {ViewProps, ViewStyle} from 'react-native';
import {cleanup, render, screen} from '@testing-library/react-native';
import '@testing-library/jest-native';
import {DefaultCard} from '../Default';
import {Typography} from '../../../../typography';
import {color} from '../../../../../themes/colors';

afterEach(cleanup);

const defaultCardParams = {isDisabled: false, onPress: () => null};
const testText = 'Hello World';
const parentId = 'parent-component';
const childId = 'child-component';

const DefaultCardComponent = (props: Partial<typeof defaultCardParams & ViewProps & ViewStyle>) => (
  <DefaultCard {...defaultCardParams} {...props} testID={parentId}>
    <Typography testID={childId}>{testText}</Typography>
  </DefaultCard>
);

describe('Default Card Component', () => {
  it('should match the existing snapshot', () => {
    const snapshot = render(<DefaultCardComponent />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should be able to render child component', () => {
    const {getByTestId} = render(<DefaultCardComponent />);

    expect(getByTestId(childId)).toBeVisible();
    expect(getByTestId(childId)).toHaveTextContent(testText);
  });

  it('should not be tappable when disabled', () => {
    const {getByTestId} = render(<DefaultCardComponent testID={parentId} isDisabled={true} />);

    expect(getByTestId(parentId)).toBeDisabled();
  });

  it('should be able to override default styling', () => {
    const newStyle = {
      backgroundColor: 'red',
    };

    render(<DefaultCardComponent {...newStyle} />);

    const parentComponent = screen.getByTestId(parentId);

    expect(parentComponent).not.toHaveStyle({backgroundColor: color.background.white});
    expect(parentComponent).toHaveStyle({...newStyle});
  });
});
