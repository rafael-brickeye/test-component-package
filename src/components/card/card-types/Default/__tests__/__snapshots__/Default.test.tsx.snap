// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Card Component should match the existing snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "elevation": 1,
      "minHeight": 68,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
    }
  }
  testID="parent-component"
>
  <View
    style={Object {}}
  >
    <Text
      numberOfLines={0}
      style={
        Object {
          "color": "#000000DE",
          "fontSize": 12,
          "fontWeight": "normal",
        }
      }
      testID="child-component"
    >
      Hello World
    </Text>
  </View>
</View>
`;
